src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -2,48 +2,17 @@
 
 import org.skypro.skyshop.product.Product;
 
-import java.util.Arrays;
-import java.util.Objects;
+import java.util.*;
 
 public class ProductBasket {
-    private Product[] products;
+    private LinkedList<Product> products;
```

Comment:
При объявлении переменной используем максимально возможный уровень абстракции 

```suggestion
    private List<Product> products;
```